internal/restrictions/update/update.go

```diff
@@ -4,44 +4,109 @@ import (
 	"context"
 	"fmt"
 	"net"
+	"slices"
 
 	"github.com/go-errors/errors"
 	"github.com/supabase/cli/internal/utils"
 	"github.com/supabase/cli/pkg/api"
 )
 
-func Run(ctx context.Context, projectRef string, dbCidrsToAllow []string, bypassCidrChecks bool) error {
-	// 1. separate CIDR to v4 and v6
+// Run updates the network restriction lists using the provided CIDRs.
+func Run(ctx context.Context, projectRef string, dbCidrsToAllow []string, bypassCidrChecks bool, appendMode bool) error {
+	newCidrsBody, err := buildRequestBody(dbCidrsToAllow, bypassCidrChecks)
+	if err != nil {
+		return err
+	}
+
+	var body api.V1UpdateNetworkRestrictionsJSONRequestBody
+
+	if appendMode {
```

Comment:
Instead of updating the network list client side, let's use the PATCH endpoint for updating network restrictions atomically https://api.supabase.com/api/v1#tag/projects/patch/v1/projects/{ref}/network-restrictions